documentation/modules/oauth/con-oauth-authentication-client.adoc

```diff
@@ -0,0 +1,25 @@
+// Module included in the following assemblies:
+//
+// assembly-oauth.adoc
+
+[id='con-oauth-authentication-client-{context}']
+= {oauth} Kafka client configuration
+A Kafka client is configured with either:
+
+* The credentials required to obtain a valid access token from an authorization server (client ID and Secret)
+* A valid long-lived access token or refresh token, obtained using tools provided by an authorization server
+
+Credentials are never sent to the Kafka broker.
```

Comment:
This is IMHO a bit misleading since in a sense the tokens are the credentials since they are all what is needed to connect (as long as the token is valid).

Reply:
What is meant here is any secret information used to authenticate with authorization server in order to obtain the 'access token' is never sent to Kafka broker. In this way Kafka broker can not impersonate user or gain access beyond that granted by user. For example, get other access tokens issued in the name of the user. In a tightly controlled intranet environment we would often assume that Kafka brokers in our cluster are not hacked, and used to gain a deeper foothold in the network, so maybe we could just remove this potentially misleading sentence as we explain later with diagrams how Kafka Broker + OAuth2 integration works.

Reply:
I know what is meant here. But the average reader might be confused especially since other part talks about how TLS should be used to avoid highjacking the token.

Reply:
Hope this makes it clear enough:

The only information ever sent to the Kafka broker is an access token.
The credentials used to authenticate with the authorization server to obtain the access token are never sent to the broker.

When a client obtains an access token, no further communication with the authorization server is needed.

Reply:
That sounds good.